lib/glimesh_web/live/user_live/stream.ex

```diff
@@ -45,36 +54,26 @@ defmodule GlimeshWeb.UserLive.Stream do
       %{}
     )
 
-    janus_uri = random_janus_server()
-    janus_hostname = hostname_from_url(janus_uri)
+    case Glimesh.Janus.get_closest_edge_location(country) do
+      %Glimesh.Janus.EdgeRoute{url: janus_url, hostname: janus_hostname} ->
+        {:noreply,
+         socket
+         |> push_event("load_video", %{
+           janus_url: janus_url,
+           channel_id: socket.assigns.channel_id
+         })
+         |> assign(:janus_url, janus_url)
+         |> assign(:janus_hostname, janus_hostname)}
 
-    {:noreply,
-     socket
-     |> push_event("load_video", %{
-       janus_uri: janus_uri,
-       channel_id: socket.assigns.channel_id
-     })
-     |> assign(:janus_uri, janus_uri)
-     |> assign(:janus_hostname, janus_hostname)}
+      _ ->
+        # In the event we can't find an edge, something is real wrong
+        {:noreply, socket}
```

Comment:
Maybe instead of just not saying what is wrong you could push something up to the user or log it and default to one edge location so they could at least view the stream

Reply:
Added an error message if this condition happens, but we can't send them a default edge location because this error only happens when there are zero available (empty database).